src/common/clipboard.ts

```diff
@@ -0,0 +1,20 @@
+export class ClipBoard<T> {
```

Comment:
We should look to standardize the clipboard usage based on the working draft of the W3C clipboard API @ https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API

I would suggest we use the standard API but fall back to a polyfill if not supported by the current browser.